libraries/ESP8266WiFi/src/ESP8266WiFiAP.cpp

```diff
@@ -85,50 +88,57 @@ static bool softap_config_equal(const softap_config& lhs, const softap_config& r
 
 /**
  * Set up an access point
- * @param ssid              Pointer to the SSID (max 31 char).
- * @param passphrase        For WPA2 min 8 char, for open use NULL (max 63 char).
+ * @param ssid              Pointer to the SSID (max 32 char).
+ * @param psk               For WPA2 min 8 char max 64 char, for open use "" or NULL.
  * @param channel           WiFi channel number, 1 - 13.
  * @param ssid_hidden       Network cloaking (0 = broadcast SSID, 1 = hide SSID)
  * @param max_connection    Max simultaneous connected clients, 0 - 8. https://bbs.espressif.com/viewtopic.php?f=46&t=481&p=1832&hilit=max_connection#p1832
  */
-bool ESP8266WiFiAPClass::softAP(const char* ssid, const char* passphrase, int channel, int ssid_hidden, int max_connection) {
+bool ESP8266WiFiAPClass::softAP(const char* ssid, const char* psk, int channel, int ssid_hidden, int max_connection) {
 
     if(!WiFi.enableAP(true)) {
         // enable AP failed
         DEBUG_WIFI("[AP] enableAP failed!\n");
         return false;
     }
 
-    if(!ssid || strlen(ssid) == 0 || strlen(ssid) > 31) {
-        // fail SSID too long or missing!
-        DEBUG_WIFI("[AP] SSID too long or missing!\n");
+    size_t ssid_len = strlen(ssid);

```

Comment:
`ssid` was tested against `nullptr` before calling `strlen()`.
same with `psk` below.
I'm not sure why this is removed and how `strlen()` would behave.

Reply:
Depends on the strlen implementation, as it turns out. idk what it is though, it is from sdk since the newlib file is commented out.
It is good to know though that newlib is a basic `while (*str) { ++len; }` :/, as it will break with null

Reply:
updated with null check which sets ...\_len to 0